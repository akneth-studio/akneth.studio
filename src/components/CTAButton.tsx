'use client'

import { Button } from "react-bootstrap";
import { useRouter } from "next/navigation";
import { ReactNode } from "react";

interface CTAButtonProps {
    text: string;
    icon?: ReactNode;
    variant?: 'primary' | 'outline-primary' | 'secondary' | 'outline-secondary' | string;
    size?: 'sm' | 'lg' | '';
    className?: string;
    to?: string;
    onClick?: () => void;
    type: 'button' | 'submit' | 'reset';
    tabIndex?: number;
    disabled?: boolean;
    "data-testid"?: string;
}

/**
 * Uniwersalny przycisk CTA do przekierowania na stronę kontaktową.
 */
export default function CTAButton({
    text,
    icon,
    to,
    onClick,
    variant = 'outline-primary',
    size = '',
    className = '',
    type = 'button',
    disabled = false,
    "data-testid": dataTestId = 'cta-button',
}: CTAButtonProps) {
    const router = useRouter();
    // Obsługa przekierowania, jeśli podano adres URL
    const handleClick = () => {
        // Jeśli jest podany onClick, użyj go
        if (onClick) {
            onClick();
        } else if (to) {
            router.push(to);
        }
    };
    return (
        <Button
            type={type}
            onClick={handleClick}
            className={`btn btn-${variant}${size ? ` btn-${size}` : ''} ${className}`}
            role="button"
            data-testid={dataTestId}
            tabIndex={0}
            aria-label={text}
            disabled={disabled}
        >
            {icon ? icon : null}
            {text}
        </Button>
    );
}