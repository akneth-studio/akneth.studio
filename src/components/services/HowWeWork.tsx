'use client'

import { Card } from 'react-bootstrap';

/**
 * Sekcja opisująca proces współpracy (Jak pracujemy)
 * @returns {JSX.Element}
 */
export default function HowWeWork() {
  return (
    <Card className="mb-5 border-primary" data-testid="how-we-work">
      <Card.Body>
        <Card.Title as="h2" className="h4 mb-4">Jak pracujemy?</Card.Title>
        <ul className="list-unstyled row">
          {[
            'Analiza potrzeb i wycena w 24h',
            'Umowa jasna i przejrzysta',
            'Płatność po odbiorze',
            'Wsparcie techniczne 3 miesiące'
          ].map((item, index) => (
            <li key={index} className="col-md-6 mb-3 d-flex">
              <span className="badge bg-primary me-2">{index + 1}</span>
              {item}
            </li>
          ))}
        </ul>
      </Card.Body>
    </Card>
  );
}
