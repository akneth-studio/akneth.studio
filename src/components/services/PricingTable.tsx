'use client'

import { Table } from 'react-bootstrap';
import { pricingData } from '@/data/pricingData';
import React from 'react';

/**
 * Tabela z cennikiem usług
 */
export default function PricingTable() {
  return (
    <div className="table-responsive mb-5" data-testid="pricing-table">
      <Table striped bordered hover className="align-middle pricing-table" lang='pl'>
        <thead className="table-dark">
          <tr>
            <th>Usługa</th>
            <th className="text-end">Zakres cen (PLN)</th>
          </tr>
        </thead>
        <tbody>
          {pricingData.map(({ id, category, services }) => (
            <React.Fragment key={id}>
              <tr className='table-secondary'>
                <td colSpan={2} className="text-center py-2">
                  <h6 className='m-0'>{category}</h6>
                </td>
              </tr>
              {services.map(({ id: serviceID, service, priceRange }) => (
                <tr key={serviceID}>
                  <td className='text-wrap align-middle'>
                    {service}
                  </td>
                  <td className="text-end ps-2" style={{ minWidth: 80 }}>
                    {priceRange}
                  </td>
                </tr>
              ))}
            </React.Fragment>
          ))}
        </tbody>
      </Table>
    </div>
  );
}