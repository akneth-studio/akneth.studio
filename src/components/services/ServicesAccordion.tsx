'use client'

import { Accordion } from "react-bootstrap";
import { servicesData } from '@/data/servicesData';
import type { CategoryType, ServiceType } from '@/lib/types';

/**
 * Komponent akordeonu z kategoriami us≈Çug
 */
export default function ServicesAccordion() {
    return (
        <Accordion defaultActiveKey="0" className="mb-5" data-testid="services-accordion">
            {servicesData.map((category: CategoryType, index: number) => (
                <Accordion.Item key={category.id} eventKey={index.toString()}>
                    <Accordion.Header>{category.name}</Accordion.Header>
                    <Accordion.Body>
                        <ul className="list-unstyled">
                            {category.services.map((service: ServiceType) => (
                                <li key={service.id} className="mb-3">
                                    <h3 className="h5">{service.title}</h3>
                                    <p>{service.description}</p>
                                </li>
                            ))}
                        </ul>
                    </Accordion.Body>
                </Accordion.Item>
            ))}
        </Accordion>
    );
}