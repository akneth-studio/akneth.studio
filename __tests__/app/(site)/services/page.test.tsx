/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import ServicesPage from '@/app/(site)/services/page';
import ServicesAccordion from '@/components/services/ServicesAccordion';
import PricingTable from '@/components/services/PricingTable';
import HowWeWork from '@/components/services/HowWeWork';
import CTAButton from '@/components/CTAButton';

// Mock the components
jest.mock('@/components/services/ServicesAccordion', () => {
  return jest.fn(() => <div data-testid="services-accordion">Services Accordion</div>);
});

jest.mock('@/components/services/PricingTable', () => {
  return jest.fn(() => <div data-testid="pricing-table">Pricing Table</div>);
});

jest.mock('@/components/services/HowWeWork', () => {
  return jest.fn(() => <div data-testid="how-we-work">How We Work</div>);
});

jest.mock('@/components/CTAButton', () => {
  return jest.fn(() => <div data-testid="cta-button">CTA Button</div>);
});

// Mock environment variables
process.env.NEXT_PUBLIC_SITE_URL = require('../../../../jest.setup').NEXT_PUBLIC_SITE_URL || 'https://akneth.pl';

describe('ServicesPage', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render all sections correctly', () => {
    render(<ServicesPage />);
    
    // Check if all sections are rendered
    expect(screen.getByTestId('services-accordion')).toBeInTheDocument();
    expect(screen.getByTestId('pricing-table')).toBeInTheDocument();
    expect(screen.getByTestId('how-we-work')).toBeInTheDocument();
    expect(screen.getByTestId('cta-button')).toBeInTheDocument();
  });

  it('should render the correct layout structure', () => {
    render(<ServicesPage />);
    
    // Check if the main container exists
    const container = screen.getByTestId('services-accordion').parentElement;
    expect(container).toBeInTheDocument();
    expect(container).toHaveClass('container');
    
    // Check if the heading exists
    const heading = screen.getByText('Nasze Usługi');
    expect(heading).toBeInTheDocument();
    expect(heading.tagName).toBe('H1');
  });

  it('should have correct metadata', () => {
    // Check if metadata is defined
    const { metadata } = require('@/app/(site)/services/page');
    expect(metadata).toBeDefined();
    expect(metadata.title).toBe('Usługi');
    expect(metadata.description).toBe('Oferta AKNETH Studio: strony internetowe, sklepy online, automatyzacje i szkolenia. Sprawdź nasze ceny i proces współpracy.');
    expect(metadata.keywords).toEqual([
      'Usługi',
      'Cennik',
      'automatyzacje',
      'szkolenia',
      'proces współpracy'
    ]);
    expect(metadata.alternates?.canonical).toBe('https://akneth.pl/services');
  });

  it('should match snapshot', () => {
    const { container } = render(<ServicesPage />);
    expect(container).toMatchSnapshot();
  });
});
