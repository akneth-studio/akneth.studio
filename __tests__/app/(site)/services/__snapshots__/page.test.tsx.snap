// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ServicesPage should match snapshot 1`] = `
<div>
  <article
    class="container mt-5"
  >
    <h1
      class="display-4 mb-4"
    >
      Nasze Us≈Çugi
    </h1>
    <div
      data-testid="services-accordion"
    >
      Services Accordion
    </div>
    <div
      data-testid="pricing-table"
    >
      Pricing Table
    </div>
    <div
      data-testid="how-we-work"
    >
      How We Work
    </div>
    <div
      class="text-center my-5"
    >
      <div
        data-testid="cta-button"
      >
        CTA Button
      </div>
    </div>
  </article>
</div>
`;
